python/openinference-instrumentation/src/openinference/instrumentation/config.py

```diff
@@ -0,0 +1,165 @@
+import os
+from dataclasses import dataclass
+from typing import Any, Optional
+
+from opentelemetry.context import (
+    _SUPPRESS_INSTRUMENTATION_KEY,
+    attach,
+    detach,
+    set_value,
+)
+
+from .logging import logger
+
+
+class suppress_tracing:
+    """
+    Context manager to pause OpenTelemetry instrumentation.
+
+    Examples:
+        with suppress_tracing():
+            # No tracing will occur within this block
+            ...
+    """
+
+    def __enter__(self) -> "suppress_tracing":
+        self._token = attach(set_value(_SUPPRESS_INSTRUMENTATION_KEY, True))
+        return self
+
+    def __aenter__(self) -> "suppress_tracing":
+        self._token = attach(set_value(_SUPPRESS_INSTRUMENTATION_KEY, True))
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback) -> None:
+        detach(self._token)
+
+    def __aexit__(self, exc_type, exc_value, traceback) -> None:
+        detach(self._token)
+
+
+OPENINFERENCE_HIDE_INPUTS = "OPENINFERENCE_HIDE_INPUTS"
+# Hides input value & messages
+OPENINFERENCE_HIDE_OUTPUTS = "OPENINFERENCE_HIDE_OUTPUTS"
+# Hides output value & messages
+OPENINFERENCE_HIDE_INPUT_MESSAGES = "OPENINFERENCE_HIDE_INPUT_MESSAGES"
+# Hides all input messages
+OPENINFERENCE_HIDE_OUTPUT_MESSAGES = "OPENINFERENCE_HIDE_OUTPUT_MESSAGES"
+# Hides all output messages
+OPENINFERENCE_HIDE_INPUT_IMAGES = "OPENINFERENCE_HIDE_INPUT_IMAGES"
+# Hides images from input messages
+OPENINFERENCE_HIDE_INPUT_TEXT = "OPENINFERENCE_HIDE_INPUT_TEXT"
+# Hides text from input messages
+OPENINFERENCE_HIDE_OUTPUT_TEXT = "OPENINFERENCE_HIDE_OUTPUT_TEXT"
+# Hides text from output messages
+OPENINFERENCE_HIDE_EMBEDDING_VECTORS = "OPENINFERENCE_HIDE_EMBEDDING_VECTORS"
+# Hides returned embedding vectors
+OPENINFERENCE_BASE64_IMAGE_MAX_LENGTH = "OPENINFERENCE_BASE64_IMAGE_MAX_LENGTH"
+# Limits characters of a base64 encoding of an image
+REDACTED_VALUE = "__REDACTED__"
+# When a value is hidden, it will be replaced by this redacted value
+
+DEFAULT_HIDE_INPUTS = False
+DEFAULT_HIDE_OUTPUTS = False
+
+DEFAULT_HIDE_INPUT_MESSAGES = False
+DEFAULT_HIDE_OUTPUT_MESSAGES = False
+
+DEFAULT_HIDE_INPUT_IMAGES = False
+DEFAULT_HIDE_INPUT_TEXT = False
+DEFAULT_HIDE_OUTPUT_TEXT = False
+
+DEFAULT_HIDE_EMBEDDING_VECTORS = False
+DEFAULT_BASE64_IMAGE_MAX_LENGTH = 32_000
+
+
+@dataclass(frozen=True)
+class TraceConfig:
+    hide_inputs: Optional[bool] = None
```

Comment:
fyi, we can leverage `metadata`
```suggestion
    hide_inputs: Optional[bool] = field(default=None, metadata={"env_var": ..., "default_value": ...}
```

Reply:
Done